### PR TITLE
Upgrade to operator-sdk v1.8.0

### PR DESCRIPTION
**Pull request by ShyamsundarR**
This is in preparation to start building bundles, which
requires moving to later sdk version.

The Makefile is changed based on initializing a dummy
sdk project and extracting and using changes from there
in the Makefile.

Rest of the changes are done as required by:
https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.8.0/

Starting from changes required to move from v1.3.0 till v1.8.0.

Signed-off-by: Shyamsundar Ranganathan <srangana@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
+#
+# For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
+# example.com/tmp-sdk-bundle:$VERSION and example.com/tmp-sdk-catalog:$VERSION.
+IMAGE_TAG_BASE ?= example.com/tmp-sdk
+
+# BUNDLE_IMG defines the image:tag used for the bundle.
+# You can use it as an arg. (E.g make bundle-build BUNDLE_IMG=<some-registry>/<project-name-bundle>:<tag>)
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
+# Default bundle image tag
+BUNDLE_IMG ?= controller-bundle:$(VERSION)
```
Comment (vdeenadh) [inline Makefile:38]: BUNDLE_IMG is being overwritten here and seems unintentional. Is it?

**Review by vdeenadh — APPROVED** [review]
LGTM. Starting from a fresh Makefile is a good idea.

File: Makefile (hunk 1 above)
Comment (ShyamsundarR) [reply Makefile]: :) Was just correcting it... had saved older value to cleanup last and did not. Corrected now.

Comment (vdeenadh) [reply Makefile]: LGTM

**Review by BenamarMk — APPROVED** [review]

**Pull request merged**